### PR TITLE
Feature/ui hydrator label as nodeid

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Moves sort-of pure functions from ```HydratorErrorFactory``` to ```non-store-pipeline-error-factory```
- Strips off uuids from nodes and connections.

##### FIXME:
- File names. This hydrator feature has terrible file names.
- ~~If there are duplicate node names (for instance two ```kvtable```) I am preventing the user from exporting or publishing the pipeline. We display a message and we need to confirm it with brady.~~
- ~~As the user adds a node the node name should be auto generated.~~

nodelabels

### CODE REVIEW

**Comment by ajainarayanan** [comment]
errors

File: cdap-ui/app/directives/dag/my-dag.html [hunk 1]
```diff
@@ -74,12 +74,12 @@
                 ng-mouseenter="MyDAGController.nodeMouseEnter(node)"
                 ng-mouseleave="MyDAGController.nodeMouseLeave(node)">
             <div class="error-node-notification"
-                 ng-if="node.error"
-                 tooltip="{{node.error.message}}"
+                 ng-if="node.errorCount > 0"
+                 tooltip="Please see node configuration panel"
```
Comment (elfenheart) [inline cdap-ui/app/directives/dag/my-dag.html]: Do we still need a tooltip then? If we are not giving specific message, I don't see a point in having a tooltip..

**Comment by elfenheart** [comment]
I think there are other places where it is still referencing node.name instead of label. For example the schema on edge does not work. Might have to publish one, and check the metrics on DAG as well.

File: cdap-ui/app/directives/dag/my-dag.html (hunk 1 above)
Comment (ajainarayanan) [inline cdap-ui/app/directives/dag/my-dag.html]: The input is from brady. Thats according to the design.

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**